learn-arai/app/register/page.tsx

```diff
@@ -0,0 +1,142 @@
+'use client';
+
+import Image from 'next/image';
+
+import { FormEvent, useState } from 'react';
+import { FaRegUser } from 'react-icons/fa';
+import { FaPhoneAlt } from 'react-icons/fa';
+import { IoMdKey } from 'react-icons/io';
+import { MdOutlineEmail } from 'react-icons/md';
+
+import Input from '@/components/module/form/input';
+
+import sideLoginPicture from '@/public/login/teaching.jpeg';
+
+import '@/app/register/register.css';
+
+export default function Home() {
+    const [message, setMessage] = useState('');
+    async function submitHandle(event: FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+
+        const formData = new FormData(event.currentTarget);
+        const result = await fetch('http://localhost:3000/auth/sign-up', {
+            method: 'POST',
+            body: formData,
+            credentials: 'include',
+        });
+        const data = await result.json();
+        if (data.status == 'error') {
+            if (data.errors) {
+                if (data.errors.password) {
+                    setMessage(data.errors.password[0]);
+                } else if (data.errors.email) {
+                    setMessage(data.errors.email[0]);
+                } else if (data.errors.passwordConfirmation[0]) {
+                    setMessage(data.errors.passwordConfirmation[0]);
+                }
+            }
+        } else {
+            window.location.href = 'checking';
+        }
+    }
```

Comment:
It would be better to move this into useAuth hook, BUT its fine "enough"

Reply:
![image](https://github.com/learn-arai/learn-arai/assets/116646446/77072ed3-cdda-46eb-8131-6f898dba87b0)